app/containers/MessageActions/index.js

```diff
@@ -193,8 +193,22 @@ const MessageActions = React.memo(forwardRef(({
 	const handleStar = async(message) => {
 		logEvent(message.starred ? events.ROOM_MSG_ACTION_UNSTAR : events.ROOM_MSG_ACTION_STAR);
 		try {
-			await RocketChat.toggleStarMessage(message.id, message.starred);
-			EventEmitter.emit(LISTENER, { message: message.starred ? I18n.t('Message_unstarred') : I18n.t('Message_starred') });
+			const db = database.active;
+			await db.action(async() => {
+				await message.update((m) => {
+					m.starred = !m.starred;
+				});
+			});
+			const result = await RocketChat.toggleStarMessage(message.id, !message.starred);
```

Comment:
If this call is rejected, database will remain on new (and wrong) state.

Reply:
Yes sir, that's why I am checking api response, and if result.success is not true then it will revert back old changes in the database which will bring database back to new state.

I am not sure if you are referring to this thing only.
Thanks :)